x-pack/plugins/cloud_security_posture/common/utils/helpers.ts

```diff
@@ -101,8 +105,10 @@ export const getStatusForIndexName = (indexName: string, status?: BaseCspSetupSt
   return 'unknown';
 };
 
-export const cleanupCredentials = (packagePolicy: NewPackagePolicy | UpdatePackagePolicy) => {
+export const cleanupAwsCredentials = (packagePolicy: NewPackagePolicy | UpdatePackagePolicy) => {
```

Comment:
My concern is regarding if this feature only needed for AWS policies. Make sense to be that if we try to clean unneeded credentials it might be something that should be applied to all policies. 

@mitodrummer @kfirpeled can you provide more input regarding that?

Reply:
@mitodrummer btw as GCP integration was introduced for 8.10 it has the same behavior as described in https://github.com/elastic/security-team/issues/6929 for `Credential` field as it has two options JSON and file name to choose from. Should I create a new ticket specifically for GCP behavior fix or do we want to use this bug as possibility to make the logic more generic and extend it to both AWS and GCP?